File: ios/Algo/Features/Auth/Models/AuthToken.swift
// AuthToken.swift
// Algo
//
// Token models for JWT-based authentication with refresh token rotation.

import Foundation

// MARK: - AuthToken

/// Response from POST /auth/login — contains access + refresh tokens and the authenticated user.
struct AuthToken: Codable, Sendable {

    let accessToken: String
    let refreshToken: String
    let expiresIn: Int
    let tokenType: String
    let user: User

    enum CodingKeys: String, CodingKey {
        case accessToken = "access_token"
        case refreshToken = "refresh_token"
        case expiresIn = "expires_in"
        case tokenType = "token_type"
        case user
    }

    /// Computes the absolute expiration date from now based on `expiresIn` seconds.
    var expirationDate: Date {
        Date().addingTimeInterval(TimeInterval(expiresIn))
    }
}

// MARK: - RefreshResponse

/// Response from POST /auth/refresh — contains a new access token only.
struct RefreshResponse: Codable, Sendable {

    let accessToken: String
    let expiresIn: Int

    enum CodingKeys: String, CodingKey {
        case accessToken = "access_token"
        case expiresIn = "expires_in"
    }
}

File: android/app/src/main/java/com/algonit/algo/features/auth/data/model/AuthToken.kt
package com.algonit.algo.features.auth.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Full authentication token response from the login/register endpoints.
 * Contains both access and refresh tokens along with user profile data.
 */
@Serializable
data class AuthToken(
    @SerialName("access_token") val accessToken: String,
    @SerialName("refresh_token") val refreshToken: String,
    @SerialName("expires_in") val expiresIn: Int,
    @SerialName("token_type") val tokenType: String,
    val user: User
)

/**
 * Response from the token refresh endpoint.
 * Only returns a new access token; the refresh token remains unchanged.
 */
@Serializable
data class RefreshResponse(
    @SerialName("access_token") val accessToken: String,
    @SerialName("expires_in") val expiresIn: Int
)
